### PR TITLE
Disable key behavior when component disabled

### PR DESCRIPTION
**Pull request by delesseps**
Also...
- Add space and enter key support for toggling checked state as per a regular checkbox
- Fix demo page not showing component
- Added tests

### CODE REVIEW

**Comment by delesseps** [comment]
@afc163  Is this suitable for merge? Thanks!

File: examples/simple.js [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 function onChange(value) {
   console.log(`switch checked: ${value}`); // eslint-disable-line
 }
 
-class Test extends React.Component {
-  state = {
-    disabled: false,
-  };
-  toggle = () => {
+const Test = React.createClass({
```
Comment (benjycui) [inline examples/simple.js]: Why??? We should not use `React.createClass` after React@15.5, just use ES6 class.

**Comment by benjycui** [comment]
👍

File: examples/simple.js (hunk 1 above)
Comment (delesseps) [inline examples/simple.js]: Np, updating was just following conventions used in other components

File: src/Switch.jsx [hunk 2]
```diff
@@ -45,11 +45,14 @@ class Switch extends Component {
   }
 
   handleKeyDown = (e) => {
-    if (e.keyCode === 37) {
-      this.setChecked(false);
-    }
-    if (e.keyCode === 39) {
-      this.setChecked(true);
+    if (!this.props.disabled) {
```
Comment (afc163) [inline src/Switch.jsx]: It is better to put `disabled` logic into `setChecked`.

Comment (delesseps) [inline src/Switch.jsx]: Ok done

**Comment by afc163** [comment]
Thx @delesseps ! `1.5.2` is released.

**Pull request merged**